File: python/1018.binary-prefix-divisible-by-5.py
class Solution:
    def prefixesDivBy5(self, nums: list[int]) -> list[bool]:

        cur = 0
        res = [False] * len(nums)

        for i in range(len(nums)):
            cur = (cur * 2 + nums[i]) % 5
            res[i] = cur == 0

        return res

File: cpp/1018.binary-prefix-divisible-by-5.cpp
#include <vector>
using std::vector;

class Solution {
public:
  vector<bool> prefixesDivBy5(vector<int> &nums) {
    int cur = 0;
    vector<bool> res;
    res.reserve(nums.size());

    for (int num : nums) {
      cur <<= 1;
      cur += num;
      cur %= 5;

      res.push_back(cur == 0);
    }

    return res;
  }
};